lib/facility.rb

```diff
@@ -1,14 +1,60 @@
 class Facility
   attr_reader :name, :address, :phone, :services
+  attr_accessor :collected_fees, :register_vehicles
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
-    @services = []
+  def initialize(attributes)
+    @name = attributes[:name]
+    @address = attributes[:address]
+    @phone = attributes[:phone]
+    @hours = attributes[:hours]
+    @services = attributes[:services] || []
+    @collected_fees = 0
+    @register_vehicles = []
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    vehicle.register
+    vehicle.set_plate_type
+    @register_vehicles << vehicle
+  end
```

Comment:
I like that this method has a bunch of helper methods in it, however, we're missing 2 pieces of logic from this method: 1. It should first be checking that the facility has the service of 'Vehicle Registration', if it doesn't, then we should not do any of these steps. 2. This method should also be calling the `calculate_fee` method I see below since upon registration, fees should be collected automatically.